src/backend/distributed/commands/truncate.c

```diff
@@ -216,7 +216,7 @@ EnsureLocalTableCanBeTruncated(Oid relationId)
 	}
 
 	/* make sure there are no foreign key references from a local table */
-	SetForeignConstraintRelationshipGraphInvalid();
+	InvalidateForeignKeyGraph();
```

Comment:
@marcocitus does this change make sense to you? Foreign key invalidation logic is a bit tough to crack for me.

@onurctirtir suggested this change https://github.com/citusdata/citus/issues/4998#issuecomment-848689857

Reply:
Or maybe we can directly use functions that are not relying on fkey cache (foreign_constarints.c)

Reply:
InvalidateForeignKeyGraph sends an invalidation via shared memory to all backends, including the current one.

However, we might not call AcceptInvalidationMessages before reading from the cache below. It would be better to add a call to AcceptInvalidationMessages in IsForeignConstraintRelationshipGraphValid.

Reply:
I plan to have 3 PRs in total:
1. This one will only change the tests
2. I will start to use functions that are not relying on fkey cache (in foreign_constarints.c) and verify that the test results are still the same
3. A small PR that will add a call to AcceptInvalidationMessages in IsForeignConstraintRelationshipGraphValid. (This functions will not be used in the PRs 1 and 2. Hence it makes sense to separate it)